lib/src/main/java/com/evervault/services/CachedOutboundRelayConfigService.java

```diff
@@ -0,0 +1,56 @@
+package com.evervault.services;
+
+import com.evervault.contracts.IProvideDecryptionAndIgnoreDomains;
+import com.evervault.contracts.IProvideOutboundRelayConfigFromHttpApi;
+import com.evervault.exceptions.HttpFailureException;
+import com.evervault.models.OutboundRelayConfigResult;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class CachedOutboundRelayConfigService implements IProvideDecryptionAndIgnoreDomains {
+
+    private static final Object lock = new Object();
+
+    private static OutboundRelayConfigResult cachedConfig;
+
+    public static void clearCache() {
+        cachedConfig = null;
+    }
+
+    private String[] alwaysIgnoreDomains;
+
+    public CachedOutboundRelayConfigService(IProvideOutboundRelayConfigFromHttpApi httpHandler, String evervaultApiUrl, String[] alwaysIgnoreDomains)
+            throws HttpFailureException, IOException, InterruptedException {
+        this(httpHandler, evervaultApiUrl, new ExecutableSchedulerService(1), alwaysIgnoreDomains);
+    }
+
+    private CachedOutboundRelayConfigService(IProvideOutboundRelayConfigFromHttpApi httpHandler, String evervaultApiUrl, ExecutableSchedulerService executableSchedulerService, String[] alwaysIgnoreDomains)
+            throws HttpFailureException, IOException, InterruptedException {
+        this.alwaysIgnoreDomains = alwaysIgnoreDomains;
+        if (cachedConfig == null) {
+            synchronized (lock) {
+                if (cachedConfig == null) {
+                    cachedConfig = httpHandler.getOutboundRelayConfig(evervaultApiUrl);
+                    executableSchedulerService.schedule(() -> {
+                        cachedConfig = httpHandler.getOutboundRelayConfig(evervaultApiUrl);
+                        return cachedConfig;
+                    }, 2, 2, TimeUnit.MINUTES);
```

Comment:
Currently polls the API every 2 minutes. This matches what we currently do in the Node.js SDK. Happy to reduce the time interval.